File: web/src/api/changelog.ts
import api from '@/lib/api'
import type { Result } from '@/types'

export interface ChangelogItem {
    id: number
    version: string
    title: string
    content: string
    stageNo?: string
    stageLabel?: string
    roadmapStatus?: 'released' | 'building' | 'planned' | string
    highlights?: string
    actionPath?: string
    upgradeEnabled?: number
    upgradeUrl?: string
    timestamp: string
    status: number
    sortOrder: number
    createdAt?: string
    updatedAt?: string
}

export const changelogApi = {
    /**
     * Song：获取已发布的发展历程列表
     */
    getList() {
        return api.get<any, Result<ChangelogItem[]>>('/changelog/list')
    },

    /**
     * Song：获取后台发展历程配置列表 (管理员)
     */
    getAdminList() {
        return api.get<any, Result<ChangelogItem[]>>('/admin/changelog/list')
    },

    /**
     * Song：新增发展历程 (管理员)
     */
    create(data: Partial<ChangelogItem>) {
        return api.post<any, Result<ChangelogItem>>('/changelog', data)
    },

    /**
     * Song：更新发展历程 (管理员)
     */
    update(id: number, data: Partial<ChangelogItem>) {
        return api.put<any, Result<ChangelogItem>>(`/changelog/${id}`, data)
    },

    /**
     * Song：删除发展历程 (管理员)
     */
    delete(id: number) {
        return api.delete<any, Result<void>>(`/changelog/${id}`)
    }
}

File: src/main/java/com/campus/trend/campus_pulse/entity/Changelog.java
package com.campus.trend.campus_pulse.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * Song：发展历程 / 版本日志表
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("sys_changelog")
public class Changelog {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String version;

    private String title;

    private String content;

    /**
     * Song：路线图阶段编号，如 "01"
     */
    private String stageNo;

    /**
     * Song：路线图阶段标签，如 "已上线"、"建设中"、"下一阶段"
     */
    private String stageLabel;

    /**
     * Song：上线状态：released / building / planned
     */
    private String roadmapStatus;

    /**
     * Song：卡片底部高亮信息，如 "发帖/评论/标签"
     */
    private String highlights;

    /**
     * Song：后续在线升级或详情入口，当前仅作为扩展配置保留
     */
    private String actionPath;

    private Integer upgradeEnabled;

    private String upgradeUrl;

    /**
     * Song：时间标签，如 "2026-02-26"
     */
    private String timestamp;

    private Integer status;

    private Integer sortOrder;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}
